### PR TITLE
Bridge DApp styling

### PR DESCRIPTION
**Pull request by koteakhaladze**

### CODE REVIEW

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { initializeHelpers, handleError, getWeb3, scaleUpUnits, scaleDownUnits,
 import Web3 from 'web3';
 import Select from 'react-select';
 
+const bigInt = require('big-integer');
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:13]: Put on line above.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 2]
```diff
@@ -117,16 +120,17 @@ export default class Bridge extends Component {
           throw new Error(`Network ID ${networkId} unsupported.`);
       }
 
+      const availableAmount = bigInt(await schnoodle.methods.unlockedBalanceOf(selectedAddress).call());
       await initializeHelpers(await schnoodle.methods.decimals().call());
 
       await this.changeTargetNetwork({ value: localStorage.getItem('targetNetwork') ?? sourceNetwork });
       await this.changeSourceNetwork({ value: localStorage.getItem('sourceNetwork') ?? sourceNetwork });
-
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:124]: Reinstate line break.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 3]
```diff
@@ -397,35 +410,44 @@ export default class Bridge extends Component {
...
+                  <input type="number" min="1" max={amount} placeholder={`Max: ${availableAmount}`} value={amount || ''} onChange={this.updateAmount} className="depositinput" />
+                  <button type="button" className="dwmbutton hidesmmd" onClick={() => this.setAmount(availableAmount / 4)}>25%</button>
+                  <button type="button" className="dwmbutton hidesmmd" onClick={() => this.setAmount(availableAmount / 2)}>50%</button>
+                  <button type="button" className="dwmbutton hidesmmd" onClick={() => this.setAmount(availableAmount * 3 / 4)}>75%</button>
+                  <button type="button" className="dwmbutton hidelg" onClick={() => this.setAmount(availableAmount / 4)}>&frac14;</button>
+                  <button type="button" className="dwmbutton hidelg" onClick={() => this.setAmount(availableAmount / 2)}>&frac12;</button>
+                  <button type="button" className="dwmbutton hidelg" onClick={() => this.setAmount(availableAmount * 3 / 4)}>&frac34;</button>
+                  <button type="button" className="maxbuttons" onClick={() => this.maxDepositAmount(availableAmount)}>Max</button>
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:421]: Have you tested this?

**Review by NeoXtreem — CHANGES_REQUESTED** [review]

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 4]
```diff
@@ -57,6 +58,8 @@ export default class Bridge extends Component {
     this.changeSourceNetwork = this.changeSourceNetwork.bind(this);
     this.changeTargetNetwork = this.changeTargetNetwork.bind(this);
     this.updateAmount = this.updateAmount.bind(this);
+    this.setAmount = this.setAmount.bind(this);
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:61]: Are you sure that you need this binding?

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 5]
```diff
@@ -10,6 +10,8 @@ import { initializeHelpers, handleError, getWeb3, scaleUpUnits, scaleDownUnits,
 import Web3 from 'web3';
 import Select from 'react-select';
 
+
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:13]: Remove line break.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx (hunk 1 above)
Comment (NeoXtreem) [reply SchnoodleDApp/ClientApp/src/components/Bridge.jsx:12]: @koteakhaladze I said put on line above, not remove lines above. Third-party libraries are grouped together as per the comment.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 6]
```diff
@@ -233,6 +236,16 @@ export default class Bridge extends Component {
 
     this.setState({ busySwap: false });
   }
+  
+  async setAmount(amount) {
+    this.setState({ amount: amount });
+  }
+
+  async updateAmount(e) {
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:244]: @koteakhaladze Have you tested if this function is even called at runtime?

**Review by NeoXtreem — CHANGES_REQUESTED** [review]
Have you actually tested your changes locally? There are still styling issues and what appears to be a major bug that will cause the values set by the buttons to not even work on the bridge.

1. Large border between the title and the top of the page (compared to the Farming page).
2. Width of textbox and SEND button too wide (compared to the Farming page).
3. Max placeholder shows an amount 18 orders of magnitude than the TS. The buttons also create such figures. This will fail on the bridge.

See comparison of Bridge and Farming page below (also shows bug 3 above).

image
image

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx (hunk 6 above)
Comment (koteakhaladze) [reply SchnoodleDApp/ClientApp/src/components/Bridge.jsx:244]: @NeoXtreem  yes this function works properly.

Comment (NeoXtreem) [reply SchnoodleDApp/ClientApp/src/components/Bridge.jsx:244]: @koteakhaladze Well, it does now that you removed the other same-named function in your [recent commit](https://github.com/schnoodledao/contracts/pull/12/commits/c306f41ae09392faec5f65d5ebf3ed8b0c3c400a). At the time you made this comment, it wasn't being called at all.

Also, why have you moved this function from its original location? It's now in an illogical place in the file.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 7]
```diff
@@ -233,6 +236,16 @@ export default class Bridge extends Component {
 
     this.setState({ busySwap: false });
   }
+  
+  async setAmount(amount) {
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:240]: Please position this function to be consistent with the Farming page.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 8]
```diff
@@ -233,6 +236,16 @@ export default class Bridge extends Component {
...
+    this.setState({ amount: scaleDownUnits(amount) });
+  }
+
+  async updateAmount(e) {
+    const value = Number(e.target.value);
+    if (!Number.isInteger(value)) return;
+    this.setState({amount: Math.min(Math.floor(value), scaleDownUnits(this.state.availableAmount))});
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:247]: This now breaks DRY principle. Why not just call `setAmount` which already sets the state?

**Review by NeoXtreem — CHANGES_REQUESTED** [review]

**Comment by NeoXtreem** [comment]
> Have you actually tested your changes locally? There are still styling issues and what appears to be a major bug that will cause the values set by the buttons to not even work on the bridge.
> 
> 1. Large border between the title and the top of the page (compared to the Farming page).
> 2. Width of textbox and SEND button too wide (compared to the Farming page).
> 3. Max placeholder shows an amount 18 orders of magnitude than the TS. The buttons also create such figures. This will fail on the bridge.
> 
> See comparison of Bridge and Farming page below (also shows bug 3 above).
> 
> image image

@koteakhaladze Item 2 above is still an issue. Please fix.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 9]
```diff
@@ -304,25 +322,54 @@ export default class Bridge extends Component {
...
 
-  render() {
-    const { networkId, sourceNetwork, targetNetwork, selectedAddress, busySwap, busyReceive, tokensPending, fee, amount, serverStatus, serverError, message } = this.state;
+  async setAmount(amount) {
+    console.log(amount);
+    this.setState({ amount: Math.min(Math.floor(amount), scaleDownUnits(this.state.availableAmount)) });
+  }
 
+  render() {
+    const { sourceNetwork, targetNetwork, selectedAddress, busySwap, busyReceive, tokensPending, fee, amount, serverStatus, serverError, message } = this.state;
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:336]: @koteakhaladze Add line break after this line (as it was previously).

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 10]
```diff
@@ -304,25 +322,54 @@ export default class Bridge extends Component {
...
-    this.setState({ amount });
+    const value = Number(e.target.value);
+    if (!Number.isInteger(value)) return;
+    this.setAmount(value);
   }
 
-  render() {
-    const { networkId, sourceNetwork, targetNetwork, selectedAddress, busySwap, busyReceive, tokensPending, fee, amount, serverStatus, serverError, message } = this.state;
+  async setAmount(amount) {
+    console.log(amount);
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:331]: Remove console log.

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 11]
```diff
@@ -304,25 +322,54 @@ export default class Bridge extends Component {
...
     const targetNetworks = sourceNetworks;
 
     const styles = {
-      valueContainer: () => ({ width: 60 }),
-      singleValue: base => ({ ...base, color: '#dc20bc' }),
-      control: (base, state) => ({ ...base, background: '#070c39', border: 'none' }),
-      dropdownIndicator: base => ({ ...base, color: '#dc20bc' }),
-      menuList: base => ({ ...base, background: '#070c39', color: '#dc20bc' }),
-      option: (provided, state) => ({ ...provided, color: state.isSelected || state.isFocused ? '#dc20bc' : '#6f1860', background: '#070c39' })
+      valueContainer: () => ({ width: 100, border: 'none', display: 'grid'}),
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:342]: Formatting (add space).

File: SchnoodleDApp/ClientApp/src/components/Bridge.jsx [hunk 12]
```diff
@@ -304,25 +322,54 @@ export default class Bridge extends Component {
...
+      singleValue: base => ({ ...base, color: 'white', }),
+      control: (base, state) => ({ ...base, background: '#070c39', borderRadius: '0.5em', borderWidth: '0px', boxShadow: '0px 12px 7px rgba(0, 0, 0, 0.34)' }),
+      dropdownIndicator: (base, state) => ({ ...base, color: 'white'}),
+      menuList: base => ({ ...base, background: '#070c39', color: 'white' }),
+      option: (provided, state) => ({ ...provided, color: state.isSelected || state.isFocused ? '#dc20bc' : 'white', background: '#070c39' })
     }
+    const { Option, SingleValue } = components;
```
Comment (NeoXtreem) [inline SchnoodleDApp/ClientApp/src/components/Bridge.jsx:349]: Add line break before.

**Review by NeoXtreem — APPROVED** [review]

**Pull request merged**